pages/docs/reference/whatsnew14.md

```diff
@@ -574,3 +574,306 @@ Complete the initial configuration, and when you add a new dependency to `cocoap
 The new dependency will be added automatically. No additional steps are required.
 
 Learn [how to add dependencies](native/cocoapods.html).
+
+
+## Standard library
+
+### Extending the common library
+
+_Common_ library is the part of the standard library that can be used in common code shared between multiple platforms.
+In 1.4, we continue extending the common library and adding or moving missing functionality to it:
+
+- `appendLine()` function that appends a given string and terminates it with `\n`:
+
+    <div class="sample" markdown="1" theme="idea" data-min-compiler-version="1.4">
+    
+    ```kotlin
+    fun main() {
+    //sampleStart
+        println(buildString {
+            appendLine("Hello,")
+            appendLine("Kotlin 1.4!")
+        })       
+    //sampleEnd
+    }
+    ```
+    </div>
+
+* `Throwable.stackTraceToString()` extension function, which returns the detailed description of this throwable with its stack trace, and `Throwable.printStackTrace()`, which prints this description to the standard error output.
+* `Throwable.addSuppressed()` function, which lets you specify the exceptions that were suppressed in order to deliver the exception, and the `Throwable.suppressedExceptions` property, which returns a list of all the suppressed exceptions.
+* `@Throws` annotation lists exception types that will be checked when the function is compiled to a platform method (on JVM or native platofrms). 
+* The common reflection API now contains only the members available on all three target platforms (JVM, JS, Native) so you can be sure that the same code works on any of them.
+
+### New functions for arrays and collections
+
+#### Collections
+
+In 1.4, the standard library includes a number of useful functions for working with **collections**:
+
+* `setOfNotNull()` makes a set consisting of all the non-null items among the provided arguments.
+
+    <div class="sample" markdown="1" theme="idea" data-min-compiler-version="1.4">
+    
+    ```kotlin
+    fun main() {
+    //sampleStart
+        val set = setOfNotNull(null, 1, 2, 0, null)
+        println(set)
+    //sampleEnd
+    }
+    ```
+    </div>
+
+* [`shuffled()`](/api/latest/jvm/stdlib/kotlin.collections/shuffled.html) is now available for sequences.
+
+    <div class="sample" markdown="1" theme="idea" data-min-compiler-version="1.4">
+    
+    ```kotlin
+    fun main() {
+    //sampleStart
+        val numbers = (0 until 50).asSequence()
+        val result = numbers.map { it * 2 }.shuffled().take(5)
+        println(result.toList()) //five random even numbers below 100
+    //sampleEnd
+    }
+    ```
+    </div>
+
+* `onEachIndexed()` and `reduceIndexedOrNull()` counterparts have been added for [onEach()](/api/latest/jvm/stdlib/kotlin.collections/on-each.html) and [reduceOrNull()](/api/latest/jvm/stdlib/kotlin.text/reduce-or-null.html), respectively.
```

Comment:
`reduceOrNull` was added as experimental in 1.3.60 or 70 and stabilized in 1.4, so it can be considered as a new function in 1.4. Therefore `reduceIndexedOrNull` is just it's indexed counterpart, not a totally new function like `onEachIndexed`